### PR TITLE
Introduce first-class editor scene bootstrap

### PR DESCRIPTION
**Pull request by kostyafarber**
## Summary

- Add EditorScene as the first-class scene owner with queryable editor.scene.value, placed glyph items, explicit placement updates, scene-local IDs, and per-item coordinate conversion.
- Replace editor route/grid bootstrap with glyph-id scene setup: one editable scene glyph at origin, without creating glyphs, sources, or layers from route load.
- Move the first rendering/coordinate consumers from global draw-offset/glyph-instance plumbing toward scene-selected placement and instance state.
- Add focused tests for route-style scene bootstrap and duplicate placements of the same GlyphId with distinct SceneGlyphIds.

## Verification

- pnpm --filter @shift/desktop lint:check passes.
- pnpm --filter @shift/desktop typecheck passes.
- pnpm --filter @shift/desktop test -- src/lib/editor/Editor.test.ts is blocked in this container because the shift-bridge native binding is missing.
- Full pre-commit hooks were not run to completion for the same environment reason: cargo/rustc are not installed, so shift-bridge cannot build its native binding here.

### CODE REVIEW

File: apps/desktop/src/renderer/src/lib/editor/EditorScene.ts [hunk 1]
```diff
@@ -0,0 +1,312 @@
+import type { Point2D } from "@shift/geo";
+import type { GlyphId } from "@shift/types";
+import type { AxisLocation } from "@/types/variation";
+import { signal, type Signal, type WritableSignal } from "@/lib/signals";
+
+declare const crypto: { getRandomValues<T extends Uint8Array>(array: T): T };
+
+const SHORT_ID_ALPHABET = "useandom-26T198340PX75pxJACKVERYMINDBUSHWOLF_GQZbfghjklqvwyzrict";
```
Comment (kostyafarber) [inline apps/desktop/src/renderer/src/lib/editor/EditorScene.ts]: do we not already have a module for this

File: apps/desktop/src/renderer/src/lib/editor/EditorScene.ts [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+
+export interface SceneGlyph {
+  readonly id: ItemId;
+  readonly kind: "glyph";
+  readonly glyphId: GlyphId;
+  readonly location: AxisLocation;
+  readonly placement: ScenePlacement;
+}
+
+export interface SceneTextRun {
```
Comment (kostyafarber) [inline apps/desktop/src/renderer/src/lib/editor/EditorScene.ts]: I don't think text runs should be its own entity

File: apps/desktop/src/renderer/src/lib/editor/EditorScene.ts [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+/**
+ * Owns the scene items visible in the editor canvas.
+ *
+ * @remarks
+ * Scene stores item identity, display location, placement, and which items
+ * show structured geometry. It does not resolve glyph models, glyph layers, or
+ * mutation capability; callers compose scene items with `Font`/`Glyph` when
+ * they need those concepts.
+ */
+export class EditorScene {
```
Comment (kostyafarber) [inline apps/desktop/src/renderer/src/lib/editor/EditorScene.ts]: `Scene`

**Comment by kostyafarber** [comment]
Implementation update pushed in `92a2a79d`.

Current design summary:
- `EditorScene` now stores first-class `ItemId` scene `items` plus `geometryItems` for items whose structured geometry is shown.
- Scene glyphs carry `glyphId`, `location`, and `placement`; they do not carry `editable`, `sourceId`, edit-layer state, or resolved glyph/runtime objects.
- Cross-boundary resolution lives on `Editor` convenience methods (`glyphForItem`, `instanceForItem`, `layerForItem`) and active geometry signals, not on `Scene`.
- Movement is `moveItemTo` / `moveItemBy`; geometry display is `showGeometry`, `hideGeometry`, `setGeometryItems`, and `isGeometryShown`.

Verification run in `/private/tmp/shift-pr-102`:
- `pnpm typecheck`
- `pnpm --filter @shift/desktop lint:check`
- `pnpm --filter @shift/desktop exec vitest run src/lib/editor/Editor.test.ts`
- `pnpm --filter @shift/desktop test`
- Commit hooks passed: `oxfmt`, `oxlint`, `tsgo typecheck`, `deadcode`, `vitest`

File: apps/desktop/src/renderer/src/lib/editor/EditorScene.ts [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+  placeGlyph(
+    glyphId: GlyphId,
+    options: { id?: ItemId; location: AxisLocation; origin: Point2D },
+  ): ItemId {
+    const id = options.id ?? mintItemId();
+    const item: SceneGlyph = {
+      id,
+      kind: "glyph",
+      glyphId,
+      location: copyAxisLocation(options.location),
```
Comment (kostyafarber) [inline apps/desktop/src/renderer/src/lib/editor/EditorScene.ts]: maybe axis location should have its own copy behaviour to expose to users

File: apps/desktop/src/renderer/src/lib/editor/EditorScene.ts [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+  }
+
+  /**
+   * Replaces the scene with one glyph item.
+   *
+   * @param glyphId - Document glyph identity to display.
+   * @param options - Scene identity, display location, and origin.
+   * @returns The scene item id for the placed glyph.
+   */
+  replaceWithGlyph(
```
Comment (kostyafarber) [inline apps/desktop/src/renderer/src/lib/editor/EditorScene.ts]: I don't think should be a method

File: apps/desktop/src/renderer/src/lib/editor/EditorScene.ts [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+      ...scene,
+      items: scene.items.map((item) =>
+        item.id === itemId && item.kind === "glyph"
+          ? copyItem({ ...item, ...patch, kind: "glyph" })
+          : item,
+      ),
+    });
+  }
+
+  toLocal(itemId: ItemId, scenePoint: Point2D): Point2D {
```
Comment (kostyafarber) [inline apps/desktop/src/renderer/src/lib/editor/EditorScene.ts]: should we use our `Mat` library transforms here

File: apps/desktop/src/renderer/src/lib/editor/EditorScene.ts [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+    const origin = this.item(itemId)?.placement.origin ?? { x: 0, y: 0 };
+    return { x: scenePoint.x - origin.x, y: scenePoint.y - origin.y };
+  }
+
+  toScene(itemId: ItemId, localPoint: Point2D): Point2D {
```
Comment (kostyafarber) [inline apps/desktop/src/renderer/src/lib/editor/EditorScene.ts]: same

File: apps/desktop/src/renderer/src/lib/editor/EditorScene.ts [hunk 8]
```diff
@@ -0,0 +1,312 @@
...
+    const origin = this.item(itemId)?.placement.origin ?? { x: 0, y: 0 };
+    return { x: localPoint.x + origin.x, y: localPoint.y + origin.y };
+  }
+}
+
+export function asItemId(id: string): ItemId {
```
Comment (kostyafarber) [inline apps/desktop/src/renderer/src/lib/editor/EditorScene.ts]: lets move the ID stuff into types like the others

File: apps/desktop/src/renderer/src/lib/editor/EditorScene.ts [hunk 9]
```diff
@@ -0,0 +1,312 @@
...
+  return id as ItemId;
+}
+
+export function mintItemId(): ItemId {
```
Comment (kostyafarber) [inline apps/desktop/src/renderer/src/lib/editor/EditorScene.ts]: move to types

File: apps/desktop/src/renderer/src/lib/editor/Scene.ts [hunk 10]
```diff
@@ -0,0 +1,312 @@
...
+    suffix += SHORT_ID_ALPHABET[byte & SHORT_ID_ALPHABET_MASK];
+  }
+  return `item_${suffix}` as ItemId;
+}
+
+function isSceneGlyph(item: SceneItem): item is SceneGlyph {
+  return item.kind === "glyph";
+}
+
+function copyItem<T extends SceneItem>(item: T): T {
```
Comment (kostyafarber) [inline apps/desktop/src/renderer/src/lib/editor/Scene.ts:311]: we need to think how to integrate these better into the actual items themselves rather that free floating functions

File: apps/desktop/src/renderer/src/lib/editor/Editor.ts [hunk 11]
```diff
@@ -178,6 +188,12 @@ export class Editor {
 
   #cursorEffect: Effect;
   #cameraMetricsEffect: Effect;
+  #textScenePlacementEffect: Effect;
```
Comment (kostyafarber) [inline apps/desktop/src/renderer/src/lib/editor/Editor.ts]: I don't like all these free floating signals added to the editor, we again have this global state on the editor

File: apps/desktop/src/renderer/src/lib/editor/Editor.ts [hunk 12]
```diff
@@ -214,13 +230,61 @@ export class Editor {
...
+        if (!item) return null;
+        const source = this.font.sourceAt(item.location);
+        if (!source) return null;
+        const record = this.font.recordForId(item.glyphId);
+        if (!record) return null;
+        return this.font.glyphSource(this.font.glyphHandleForName(record.name), source);
+      },
+      { name: "editor.geometry.layer" },
+    );
+    this.#geometryOrigin = computed(
```
Comment (kostyafarber) [inline apps/desktop/src/renderer/src/lib/editor/Editor.ts]: this origin might not make sense in our new scene model

File: apps/desktop/src/renderer/src/lib/editor/Editor.ts [hunk 13]
```diff
@@ -214,13 +230,61 @@ export class Editor {
...
+      () => this.#geometryItem.value?.placement.origin ?? { x: 0, y: 0 },
+      { name: "editor.geometry.origin" },
+    );
 
     this.#view = new EditorViewState();
     this.input = new EditorInput();
     this.gesture = new EditorGesture();
 
-    this.#commands = new CommandRunner(this.#glyph.edit.glyphSource);
+    this.#commands = new CommandRunner(this.#geometryLayer);
```
Comment (kostyafarber) [inline apps/desktop/src/renderer/src/lib/editor/Editor.ts]: if we need to re-work the commands we can do so in a follow up commit instead of retrofitting

File: apps/desktop/src/renderer/src/lib/editor/Editor.ts [hunk 14]
```diff
@@ -279,6 +343,28 @@ export class Editor {
       },
       { name: "editor.cursor" },
     );
+
+    this.#textScenePlacementEffect = effect(
```
Comment (kostyafarber) [inline apps/desktop/src/renderer/src/lib/editor/Editor.ts]: lets re-think text placement interactions later

File: apps/desktop/src/renderer/src/lib/editor/Editor.ts [hunk 15]
```diff
@@ -520,7 +601,19 @@ export class Editor {
   }
 
   public getActiveGlyphName(): string | null {
```
Comment (kostyafarber) [inline apps/desktop/src/renderer/src/lib/editor/Editor.ts]: we might need to rethink current editor methods in the new scene world

File: apps/desktop/src/renderer/src/lib/editor/Editor.ts [hunk 16]
```diff
@@ -548,6 +641,16 @@ export class Editor {
     const glyph = this.getGlyph(handle);
     if (!glyph) return null;
 
+    const record = this.font.recordForName(handle.name);
+    if (record) {
+      const itemId = this.scene.replaceWithGlyph(record.id, {
+        id: asItemId("main"),
```
Comment (kostyafarber) [inline apps/desktop/src/renderer/src/lib/editor/Editor.ts]: we should not have to this asItemId stuff

Comment (kostyafarber) [reply apps/desktop/src/renderer/src/lib/editor/Editor.ts]: we mint it or at least prefix the mint

File: apps/desktop/src/renderer/src/lib/editor/Editor.ts [hunk 17]
```diff
@@ -938,13 +1090,11 @@ export class Editor {
   }
 
   public sceneToGlyphLocal(point: Point2D): Point2D {
-    const offset = this.drawOffset;
-    return { x: point.x - offset.x, y: point.y - offset.y };
+    return this.scene.toLocal(this.requireItemId(), point);
```
Comment (kostyafarber) [inline apps/desktop/src/renderer/src/lib/editor/Editor.ts]: requireItemId?

File: apps/desktop/src/renderer/src/lib/editor/Editor.ts [hunk 18]
```diff
@@ -1277,18 +1429,25 @@ export class Editor {
   }
 
   public get drawOffset(): Point2D {
-    return this.#text.drawOffset.peek();
+    return this.#geometryOrigin.peek();
   }
 
   /** @knipclassignore */
   public get $drawOffset(): Signal<Point2D> {
```
Comment (kostyafarber) [inline apps/desktop/src/renderer/src/lib/editor/Editor.ts:1295]: this probably does not make sense anymore in the scene world

File: apps/desktop/src/renderer/src/lib/editor/Editor.ts [hunk 19]
```diff
@@ -1301,3 +1460,7 @@ export class Editor {
     return this.#toolState[scope];
   }
 }
+
+function itemIdForTextAnchor(anchor: GlyphAnchor): ItemId {
```
Comment (kostyafarber) [inline apps/desktop/src/renderer/src/lib/editor/Editor.ts]: text anchor concept might not be needed anymore

File: apps/desktop/src/renderer/src/lib/editor/rendering/overlays/Handles.ts [hunk 20]
```diff
@@ -15,48 +16,29 @@ import { CanvasHandleRenderer } from "./handles/CanvasHandleRenderer";
...
     this.#markerLayer = layer;
     this.#markers.resetUpload();
   }
 
-  draw(canvas: Canvas, camera: CameraTransform, drawOffset: Point2D): void {
-    const list = this.#displayList.peek();
+  draw(
+    canvas: Canvas,
+    camera: CameraTransform,
+    drawOffset: Point2D,
```
Comment (kostyafarber) [inline apps/desktop/src/renderer/src/lib/editor/rendering/overlays/Handles.ts:33]: do we need a drawOffset? I would think whats coming in has everythinr we need.
